crates/host/src/wasmbus/mod.rs

```diff
@@ -3050,6 +3149,49 @@ impl Host {
         let spec: ComponentSpecification = serde_json::from_slice(value.as_ref())
             .context("failed to deserialize component specification")?;
 
+        // Compute all new links that do not exist in the host map, which we'll use to
```

Comment:
Might be good to note here that this is a tradeoff between locking for longer and allocations. We could do less allocations, but have to hold two read locks. Good for our future selves if we ever need to optimize